packages/delivery-react-native/test/delivery.test.js

```diff
@@ -1,25 +1,41 @@
 /* global describe, it, expect */
 
 const Client = require('@bugsnag/core/client')
-const VALID_NOTIFIER = { name: 't', version: '0', url: 'http://' }
 const delivery = require('../')
 
 describe('delivery: react native', () => {
-  it('sends using the native dispatch() method', done => {
+  it('sends the correct payload using the native client’s dispatch() method', done => {
+    const sent = []
     const NativeClient = {
-      dispatch: () => new Promise((resolve) => resolve(1))
+      dispatch: (event) => {
+        sent.push(event)
+        return new Promise((resolve) => resolve(true))
+      }
     }
-    const c = new Client(VALID_NOTIFIER)
-    c.setOptions({ apiKey: 'api_key' })
-    c.configure()
-    c.delivery(client => delivery(client, NativeClient))
-    const threads = [{}, {}]
-    c.notify(new Error('123'), report => {
-      report.set('threads', threads)
-    }, (err, report) => {
+    const c = new Client({ apiKey: 'api_key' })
+    c._setDelivery(client => delivery(client, NativeClient))
+    c.leaveBreadcrumb('hi')
+    c.setContext('test screen')
+    c.setUser('123')
+    c.notify(new Error('oh no'), (e) => {
+      e.groupingHash = 'ER_GRP_098'
+    }, (err, event) => {
       expect(err).not.toBeTruthy()
-      expect(report.get('errorMessage')).toBe('123')
-      expect(report.get('threads')).toEqual(threads)
+      expect(sent.length).toBe(1)
+      expect(sent[0].errors[0].errorMessage).toBe('oh no')
+      expect(sent[0].severity).toBe('warning')
+      expect(sent[0].severityReason.type).toBe('handledException')
+      expect(sent[0].unhandled).toBe(false)
+      expect(sent[0].app).toEqual({ releaseStage: 'production', version: null, type: null })
+      expect(sent[0].device).toEqual({})
+      // TODO enable once event.threads exists
```

Comment:
assuming that this will be addressed as part of a later PR

Reply:
Yup, we need too add `threads` to JS's `event` interface but I'll do that in a different PR.